### PR TITLE
Updated precommit and prepush notifications

### PR DESCRIPTION
**Pull request by ghost**
## Type of change

- [X] Bug Fix
- [ ] New Feature
- [X] Refactor
- [ ] Project Build
- [ ] Project Configuration

## Submission Checklist

- [x] Have you followed the guidelines in our [Contributing document](https://github.com/fanappics/buoy/blob/master/CONTRIBUTING.md)?
- [ ] Did you create appropriate tests?

## Description
- Remove default example usage to encourage proper documentation
- Fixes occasional change miss: `==` and `-eq` work differently
- Add a PR template notification for prepush
- Typo fix
- button margin fix

## Related Issues / Bugs / Feature Requests

https://jira.fanatics.com/browse/CPDT-164

## Test Deployment URL

Replace branch-name with the name of your branch.

https://rawgit.com/fanappics/buoy/ft-CPDT-164/docs/index.html

## Steps to Test or Reproduce

1. Building styleguide works
1. Running styleguide works
1. Commits fail if styleguide is not rebuild after components change
1. Push message notifies about PR template
1. BButtons have margins

### CODE REVIEW

**Comment by kierachell** [comment]
In order to have CircleCI push to gh pages and master different builds I would need to add a `machine user` which sounds like a bit too much work for little gain.
This is a middle ground solution that will at least inform the developers and help out a bit with the PR process:
screen shot 2018-02-02 at 8 22 31 am

**Comment by kierachell** [comment]
Fixed some Styleguide examples from inline to <componentname.md>
screen shot 2018-02-02 at 8 54 42 am

**Review by thedevelopnik — DISMISSED** [review]

File: src/components/BButton/BButton.vue [hunk 1]
```diff
@@ -81,5 +81,8 @@ button {
   background-color: #00AAED;
   border-color: #00AAED;
   color: #FFFFFF;
+  margin-bottom: 1.25rem;
```
Comment (eleanormal) [inline src/components/BButton/BButton.vue]: This should not be part of this PR

File: src/components/BButton/BButton.vue [hunk 2]
```diff
@@ -81,5 +81,8 @@ button {
...
+  padding-left: 0.9375rem;
+  padding-right: 0.9375rem;
 }
```
Comment (ghost-) [inline src/components/BButton/BButton.vue]: There is pushback against this.  You should discuss with @eleanormal, Ryan and @cochoa89 and see if we need to remove this code.  Its a bit out of scope compared to the other work though.  I think it shoiuld be at least in its own PR.

**Review by elemjay19 — DISMISSED** [review]

**Review by AnalogGhost — CHANGES_REQUESTED** [review]

**Review by AnalogGhost — DISMISSED** [review]

**Review by elemjay19 — DISMISSED** [review]

File: build/styleguide.config.js [hunk 3]
```diff
@@ -11,21 +12,20 @@ const extractPlugin = ExtractTextPlugin.extract({
...
       devServer: {
 	contentBase: resolve('../docs'),
-	publicPath: '/dev/',
+	publicPath: '/docs/',
 	disableHostCheck: true
       }
     }),
     mixins: [
-      '../src/mixins/styleguide/styleguidedist.js'
+	'../src/mixins/styleguide/styleguidist.js'
```
Comment (stevenmwade) [inline build/styleguide.config.js]: Is this purposefully spelled wrong?

Comment (kierachell) [reply build/styleguide.config.js]: https://github.com/vue-styleguidist/vue-styleguidist 🤷‍♂️

**Review by thedevelopnik — APPROVED** [review]

**Review by AnalogGhost — APPROVED** [review]
